Add successful and unsuccessful events for lever posting apply

mobinzk opened the pull request.

mobinzk committed:
Add successful and unsuccessful events for lever posting apply

peterkeating commented on Api/Services/PostingApiService.cs:
```diff
@@ -85,10 +90,41 @@ public async Task<PostingResult> Apply(LeverSettings settings, LeverPostingApply
+
+                if (json.Ok)
+                {
+                    await _workflowManager.TriggerEventAsync(
```
I'd suggest adding a method for the trigger of an event as the only differences between line 97 and 115 is the value of `IsSuccessful` (e.g. `TriggerNotificationEvent(false)` where false represents the value of `IsSuccessful`.

peterkeating commented on Api/Services/PostingApiService.cs:
```diff
@@ -85,10 +90,41 @@ public async Task<PostingResult> Apply(LeverSettings settings, LeverPostingApply
+
+                return json;
             }
             catch (Exception e)
```
Would including the exception message within the payload sent to the workflow event be useful?

peterkeating reviewed (commented):
Couple of bits from me and a handful of code smells from Sonar that would be good to resolve - https://sonarcloud.io/project/issues?id=Etch.OrchardCore.Lever&pullRequest=12&resolved=false&types=CODE_SMELL

mobinzk committed:
Remove duplicate codes

- Remove duplicate codes.
- Include exception error message in payload
- Fix code smells

mobinzk committed:
Add missing variable
